test/infamy/restconf.py

```diff
@@ -26,16 +26,19 @@ class Location:
 
 def xpath_to_uri(xpath, extra=None):
     """Convert xpath to HTTP URI"""
-    # If the xpath has a
+    # Handle [.='value'] syntax for leaf-list items
     pattern = r'\[(.*?)=["\'](.*?)["\']\]'
     matches = re.findall(pattern, xpath)
 
+    uri_path = xpath
     if matches:
         for key, value in matches:
-            # replace [key=value] with =value
-            uri_path = re.sub(rf'\[{re.escape(key)}=["\']{re.escape(value)}["\']\]', f'={value}', xpath)
-    else:
-        uri_path = xpath
+            if key == '.':
+                # For [.='value'], replace with =value
+                uri_path = re.sub(rf'\[\.=["\']{re.escape(value)}["\']\]', f'={value}', uri_path)
+            else:
+                # For [key='value'], replace with =value
+                uri_path = re.sub(rf'\[{re.escape(key)}=["\']{re.escape(value)}["\']\]', f'={value}', uri_path)
```

Comment:
```suggestion
    uri_path = re.sub(r"\[(.|\w+)='(\w+)'\]", r"=\2", xpath)
```

Not sure if I understand everything that is going on here, but is there any case that the current implementation handles, that would not covered by this suggestion?

Reply:
Nice point, I got lost in regexps trying to fix this one.  The actual bug was that we used xpath for each key,value match, instead of the converted value from any previous call in the for loop.